src/algebra/module.lean

```diff
@@ -358,12 +358,27 @@ end ideal
 
 section prio
 set_option default_priority 100 -- see Note [default priority]
+
+/- Note[vector space definition]:
```

Comment:
```suggestion
/-! Note[vector space definition]:
```
Should such notes be made module docs so they show up in the mathlib doc pages?

Reply:
With other notes we haven't done this so far. @robertylewis Can you say what's good here?

Reply:
I wouldn't do this for now. Since referencing a note won't link to it, I think it will just look out of place.

The better system is to have a command `implementation_note "vector space definition" "..."` or something. This will store the note in the environment, probably as a declaration with an attribute. Then the doc generator can retrieve all the notes and print them, either inline where they're declared or in a separate file (or both). We'll make sure the syntax for referencing a note can be matched with a regex, then all references can be linked.

In principle we don't need the command. If `Note[...]` is uniquely used for declaring notes, then `/-! Note[...]` would be enough. But I think the command is cleaner.